File: benchmarks/pandas/bench_dataframe_dropna.py
"""Benchmark: dataframe_dropna — drop rows with NaN values from 100k-row DataFrame"""
import json, time
import numpy as np
import pandas as pd

ROWS = 100_000
WARMUP = 5
ITERATIONS = 20

a = np.where(np.arange(ROWS) % 10 == 0, np.nan, np.arange(ROWS) * 1.1)
b = np.where(np.arange(ROWS) % 7 == 0, np.nan, np.arange(ROWS) * 2.2)
df = pd.DataFrame({"a": a, "b": b})

for _ in range(WARMUP):
    df.dropna()

start = time.perf_counter()
for _ in range(ITERATIONS):
    df.dropna()
total = (time.perf_counter() - start) * 1000

print(json.dumps({
    "function": "dataframe_dropna",
    "mean_ms": total / ITERATIONS,
    "iterations": ITERATIONS,
    "total_ms": total,
}))

File: benchmarks/tsb/bench_dataframe_dropna.ts
/**
 * Benchmark: dataframe_dropna — drop rows with NaN values from 100k-row DataFrame
 */
import { DataFrame } from "../../src/index.js";

const ROWS = 100_000;
const WARMUP = 5;
const ITERATIONS = 20;

const a = Float64Array.from({ length: ROWS }, (_, i) => (i % 10 === 0 ? NaN : i * 1.1));
const b = Float64Array.from({ length: ROWS }, (_, i) => (i % 7 === 0 ? NaN : i * 2.2));
const df = new DataFrame({ a, b });

for (let i = 0; i < WARMUP; i++) {
  df.dropna();
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  df.dropna();
}
const total = performance.now() - start;

console.log(
  JSON.stringify({
    function: "dataframe_dropna",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);
